ocpmodels/preprocessing/atoms_to_graphs.py

```diff
@@ -0,0 +1,186 @@
+        self.struct = AseAtomsAdaptor.get_structure(self.atoms)
+        # these return jagged arrays meaning certain atoms have more neighbors than others
+        _c_index, n_index, _images, n_distance = self.struct.get_neighbor_list(
+            r=self.radius
+        )
+        # find the delimiters, the number of neighbors varies
+        delim = np.where(np.diff(_c_index))[0] + 1
+        # split the neighbor distance and neighbor index based on delimiter
+        self.split_n_distances = np.split(n_distance, delim)
+        self.split_n_index = np.split(n_index, delim)
+
+    def _pad_arrays(self):
+        # c_index is the center index, the atom to find the neighbors of
+        c_index = np.arange(0, len(self.atoms), 1)
+        # pad c_index
+        self.all_c_index = np.repeat(c_index, self.max_neigh)
+        # add dummy variables to desired array length
+        self.all_n_index = np.full(
+            (len(self.atoms), self.max_neigh), self.dummy_index
+        )
+        self.all_distances = np.full(
+            (len(self.atoms), self.max_neigh), float(self.dummy_distance)
+        )
+
+        # loop over the stucture and replace dummy variables where values exist
+        for i, (n_index, distances) in enumerate(
+            zip(self.split_n_index, self.split_n_distances)
+        ):
+            if len(distances) == self.max_neigh:
+                self.all_n_index[i] = n_index
+                self.all_distances[i] = distances
+                continue
+            # padding arrays
+            elif len(distances) < self.max_neigh:
+                n_len = len(distances)
+                # potentially add if n_len == 0: print (increase radius)
+                self.all_n_index[i][:n_len] = n_index
+                self.all_distances[i][:n_len] = distances
+                continue
+            # removing extra values so the length is equal to max_neigh
+            # values are sorted by distance so only nearest neighbors are kept
+            elif len(distances) > self.max_neigh:
+                # this sorts the list min -> max and returns the indicies
+                sorted_dist_i = np.argsort(distances)
+                self.all_n_index[i] = n_index[sorted_dist_i[: self.max_neigh]]
+                self.all_distances[i] = distances[
+                    sorted_dist_i[: self.max_neigh]
+                ]
+
+    def _gen_features(self):
+        # expand distances in gaussian basis
+        gaussian_distances = expand_distance_gauss(self.all_distances)
+        # reshape the array
+        self.gauss_distances = gaussian_distances.reshape(
+            (
+                gaussian_distances.shape[0] * gaussian_distances.shape[1],
+                gaussian_distances.shape[2],
+            )
+        )
+        # convert to torch tensor, gauss_distances == edge_attr
+        self.edge_attr = torch.FloatTensor(self.gauss_distances)
+        # One-hot encoding for atom type embeddings
+        embeddings = np.vstack(
```

Comment:
There was a discussion that @abhshkdz brought up about these, such that a general embedding for each element type did just as well as these hard coded ones we have in place. Do we want to explore that further?